libevmasm/Assembly.cpp

```diff
@@ -756,3 +759,25 @@ LinkerObject const& Assembly::assemble() const
 	}
 	return ret;
 }
+
+vector<size_t> Assembly::decodeSubIds(size_t _subObjectId) const
+{
+	if (_subObjectId < m_subs.size())
+		return {_subObjectId};
+
+	assertThrow(m_subObjectPaths.count(_subObjectId) > 0, AssemblyException, "");
+	return m_subObjectPaths.at(_subObjectId);
+}
+
+size_t Assembly::encodeSubIds(std::vector<size_t> const& _subIdPath)
+{
+	assertThrow(!_subIdPath.empty(), AssemblyException, "");
+	if (_subIdPath.size() == 1)
+	{
+		assertThrow(_subIdPath[0] < m_subs.size(), AssemblyException, "");
+		return _subIdPath[0];
+	}
+	assertThrow(m_currentSubObjectId >= m_subs.size(), AssemblyException, "");
+	m_subObjectPaths[m_currentSubObjectId] = _subIdPath;
```

Comment:
This adds one element per call to `encodeSubIds`. Is it possible to use `_subIdPath` as a key for a `std::map`?

Reply:
Fixed